app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/NotificationFilterActivity.java

```diff
@@ -0,0 +1,214 @@
+                    case NOTIFICATION_FILTER_MODE_WHITELIST:
+                        mEditTextWords.setEnabled(true);
+                        mSpinnerFilterSubMode.setEnabled(true);
+                        break;
+
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+
+        mSpinnerFilterSubMode.setSelection(mNotificationFilter.getNotificationFilterSubMode());
+
+        mEditTextWords = findViewById(R.id.editTextWords);
+
+        if (!mWordsList.isEmpty()) {
+            StringBuilder builder = new StringBuilder();
+            for (String temp : mWordsList) {
+                builder.append(temp);
+                builder.append("\n");
+            }
+            mEditTextWords.setText(builder.toString());
+        }
+
+        mEditTextWords.setEnabled(mSpinnerFilterMode.getSelectedItemPosition() == NOTIFICATION_FILTER_MODE_NONE);
+
+        mButtonSave = findViewById(R.id.buttonSaveFilter);
+        mButtonSave.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                // TODO: check for modifications, only save if something changed
+                String words = mEditTextWords.getText().toString();
+
+                if (StringUtils.isBlank(words) && mSpinnerFilterMode.getSelectedItemPosition() != NOTIFICATION_FILTER_MODE_NONE) {
+                    Toast.makeText(NotificationFilterActivity.this, R.string.toast_notification_filter_words_empty_hint, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                try {
+                    db = GBApplication.acquireDB();
+                    NotificationFilterDao notificationFilterDao = db.getDaoSession().getNotificationFilterDao();
+                    NotificationFilterEntryDao notificationFilterEntryDao = db.getDaoSession().getNotificationFilterEntryDao();
+
+                    debugOutput(notificationFilterDao);
+
+                    mNotificationFilter.setNotificationFilterMode(mSpinnerFilterMode.getSelectedItemPosition());
+                    mNotificationFilter.setNotificationFilterSubMode(mSpinnerFilterSubMode.getSelectedItemPosition());
+
+                    notificationFilterEntryDao.deleteByKeyInTx(mFilterEntryIds);
+
+                    Long filterId = notificationFilterDao.insertOrReplace(mNotificationFilter);
+
+                    // only save words if filter mode != none
+                    if (mNotificationFilter.getNotificationFilterMode() != NOTIFICATION_FILTER_MODE_NONE) {
+                        String[] wordsSplitted = words.split("\n");
+                        for (String temp : wordsSplitted) {
+
+                            if (StringUtils.isBlank(temp)) {
+                                continue;
+                            }
+
+                            temp = temp.trim();
```

Comment:
Wouldn't it be better to trim() before checking for isBlank()?

Reply:
isBlank() gets 'em all I think